tests/ha/single_pod_restart/test_data_pod_restart_api.py

```diff
@@ -1012,3 +1015,210 @@ def test_copy_object_during_pod_restart(self):
+            resp = self.ha_api.list_verify_version(self.s3_ver, new_bucket,
+                                                   self.version_etag[new_bucket])
+            assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Step 8: Uploaded same object %s after enabling versioning. Listed & "
+                        "verified the VersionID for the same for %s.", self.object_name, new_bucket)
+
+        LOGGER.info("Step 9: Get versions of %s with & without specifying VersionID & verify "
+                    "etags for %s.", self.object_name, new_bucket)
+        resp = self.ha_api.parallel_get_object(event=event, s3_ver_obj=self.s3_ver,
+                                               bkt_name=new_bucket, obj_name=self.object_name,
+                                               ver_etag=self.version_etag[new_bucket])
+        assert_utils.assert_true(resp[0], f"Get Object with versionID failed {resp[1]}")
+        resp = self.s3_test_obj.get_object(new_bucket, self.object_name)
+        latest_vetag = self.version_etag[new_bucket][-1]
+        latest_v = list(latest_vetag.keys())[0]
+        etag = list(latest_vetag.values())[0]
+        if resp[1]["VersionId"] != latest_v:
+            assert_utils.assert_true(False, "Get Object without specifying VersionID does not "
+                                            f"match with latest {latest_v} {resp[1]}")
+        if resp[1]["ETag"] != etag:
+            assert_utils.assert_true(False, "Etag without specifying VersionID does not match "
+                                            f"with etag {etag} {resp[1]}")
+        LOGGER.info("Step 9: Got versions of %s with & without specifying VersionID & verified "
+                    "etags for %s", self.object_name, new_bucket)
+        LOGGER.info("Step 10: Restart data pod with replica method and check cluster status")
+        resp = self.ha_obj.restore_pod(pod_obj=self.node_master_list[0],
+                                       restore_method=self.restore_method,
+                                       restore_params={"deployment_name": self.deployment_name,
+                                                       "deployment_backup":
+                                                           self.deployment_backup,
+                                                       "num_replica": self.num_replica,
+                                                       "set_name": self.set_name},
+                                       clstr_status=True)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], f"Failed to restore pod by {self.restore_method} way")
+        self.restore_pod = False
+        LOGGER.info("Step 10: Successfully restart data pod with replica method & checked "
+                    "cluster status")
+        LOGGER.info("Step 11: Get versions of %s with & without specifying VersionID & verify "
+                    "etags for %s.", self.object_name, new_bucket)
```

Comment:
if DTM disabled we want to check for new_bucket only and not the original bucket?

Reply:
> if DTM disabled we want to check for new_bucket only and not the original bucket?

Moved this step to dtm flag and kept step 14 common for before after restart version verification.